api/src/services/collections.ts

```diff
@@ -579,8 +579,10 @@ export class CollectionsService {
 
 					if (revisionsToDelete.length > 0) {
 						const keys = revisionsToDelete.map((record) => record.id);
-						for(let i = 0; i < keys.length; i += 10_000) {
-							await trx('directus_revisions').update({ parent: null }).whereIn('parent', keys.slice(i, i + 10_000));
+						for (let i = 0; i < keys.length; i += 10_000) {
```

Comment:
10000 feels like an arbitrary number that we might want to make configurable. I'm also curious how this interplays with other transaction insertions; should we do the same batched insert everywhere?

Reply:
10.000 was choosen because the limit in SQLite is 32766 and the charakter limit is 1.000.000 so to be extra sure we don't hit those limits I divided 1.000.000 / 60 charakters which is about 16.000 and to be extra sure again I went down to 10.000. But I agree that this is completely arbitrary though I'm not sure where else we do something similar / how that configuration variable would be named?